test/cpp/qps/server_async.cc

```diff
@@ -166,8 +100,16 @@ class AsyncQpsServerTest {
         return;
       }));
     }
-    while (!got_sigint) {
-      std::this_thread::sleep_for(std::chrono::seconds(5));
+  }
+  ~AsyncQpsServerTest() {
+    server_->Shutdown();
+    srv_cq_.Shutdown();
```

Comment:
Requests that have started might produce events after a shutdown initiates - but that will block the shutdown event from being produced.

New requests (via start_batch) after shutdown are illegal and will crash us.

So... there may be events from the completion queue after shutdown is called, but not after the event is produced.